src/ifcparse/IfcSchema.cpp

```diff
@@ -149,3 +158,15 @@ std::vector<std::string> IfcParse::schema_names() {
 
 	return return_value;
 }
+
+void IfcParse::clear_schemas() {
+	// Populate vector with map values
```

Comment:
Why the intermediate vector, why not a single loop and `delete pair.second`?

Reply:
currently, the destructor of schema_definition removes itself from the global `schemas`, so we need a local copy first to avoid concurrent iterating and erasing on `schemas`.